### ISSUE TITLE
HRLS should not show HS_SECKEY field 

### ISSUE THREAD

**Issue by cookie33** [EUDAT-B2HANDLE/B2HANDLE-HRLS#20]
Hi,

There might be users who have the `HS_SECKEY` field filled in in a user handle. (NOT wise).

The `retrieverecords=true` should NOT show these fields when using the hrls servlet.

**Comment by cookie33** [comment]
It also should NOT allow searching on this field.

**Comment by TobiasWeigel** [comment]
Fixed in 049c7ff.

**Comment by cookie33** [comment]
tested with new created war file:
```
curl -s -u "841:<password>" "https://epic3.storage.surfsara.nl:8001/hrls/handles?HS_SECKEY=*"
Searching via HS_SECKEY entries is not allowed!
```

`retrieverecords=true` does not show `HS_SECKEY`:
```
$ curl -s -u "841:<password>" "https://epic3.storage.surfsara.nl:8001/hrls/handles?HS_ADMIN=*&limit=10&retrieverecords=true" | python -m json.tool
{
    "0.NA/0.NA": [
        {
            "type": "HS_SITE", 
            "value": "\u0000\u0001\u0002\u0005\u0000\u0001\u0080\u0002\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0001\u0000\u0000\u0000\u0004desc\u0000\u0000\u0000\"SURFsara handle & epic test server\u0000\u0000\u0000\u0001\u0000\u0000\u0000\u0001\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0091d\u001f\u009a\u0000\u0000\u0001\u00b8\u0000\u0000\u0000\u000bDSA_PUB_KEY\u0000\u0000\u0000\u0000\u0000\u0015\u0000\u0097`P\u008f\u0015#\u000b\u00cc\u00b2\u0092\u00b9\u0082\u00a2\u00eb\u0084\u000b\u00f0X\u001c\u00f5\u0000\u0000\u0000\u0081\u0000\u00fd\u007fS\u0081\u001du\u0012)R\u00dfJ\u009c.\u00ec\u00e4\u00e7\u00f6\u0011\u00b7R<\u00efD\u0000\u00c3\u001e?\u0080\u00b6Q&iE]@\"Q\u00fbY=\u008dX\u00fa\u00bf\u00c5\u00f5\u00ba0\u00f6\u00cb\u009bUl\u00d7\u0081;\u0080\u001d4o\u00f2f`\u00b7k\u0099P\u00a5\u00a4\u009f\u009f\u00e8\u0004{\u0010\"\u00c2O\u00bb\u00a9\u00d7\u00fe\u00b7\u00c6\u001b\u00f8;W\u00e7\u00c6\u00a8\u00a6\u0015\u000f\u0004\u00fb\u0083\u00f6\u00d3\u00c5\u001e\u00c3\u00025T\u0013Z\u0016\u00912\u00f6u\u00f3\u00ae+a\u00d7*\u00ef\u00f2\"\u0003\u0019\u009d\u00d1H\u0001\u00c7\u0000\u0000\u0000\u0081\u0000\u00f7\u00e1\u00a0\u0085\u00d6\u009b=\u00de\u00cb\u00bc\u00ab\\6\u00b8W\u00b9y\u0094\u00af\u00bb\u00fa:\u00ea\u0082\u00f9WL\u000b=\u0007\u0082gQYW\u008e\u00ba\u00d4YO\u00e6q\u0007\u0010\u0081\u0080\u00b4I\u0016q#\u00e8L(\u0016\u0013\u00b7\u00cf\t2\u008c\u00c8\u00a6\u00e1<\u0016z\u008bT|\u008d(\u00e0\u00a3\u00ae\u001e+\u00b3\u00a6u\u0091n\u00a3\u007f\u000b\u00fa!5b\u00f1\u00fbbz\u0001$;\u00cc\u00a4\u00f1\u00be\u00a8Q\u0090\u0089\u00a8\u0083\u00df\u00e1Z\u00e5\u009f\u0006\u0092\u008bf^\u0080{U%d\u0001L;\u00fe\u00cfI*\u0000\u0000\u0000\u0080=@;\u00c7\u00d9\u0005<\u00e1}\n\u00cb\u00d7\u0018\u00ccf\u00ed\u0018\u00cc!e\u00f5\u000f&\u00ee\u00f2\u001e\u008f\u0094&T\u00b7\u00b4~l\u009b,X\u00dcc\u00a0\u00af\u00dc\u00c8H\u0002\u007f\u0081e\u0013%X\u0094\u009a\u001bg\u0015m\u00c9K(\u00cd\u00df\u00ea\u0011\u00c28\u00a0?\u00f3\u00ac\u00e4\u00c4\u00a8\u00f6Yz\u00f1XeCXj\u00a2\u00fex]t>\u0010\u00b2\u001bh\\\u00d1j\u00ab9\u00c7H\u0095\u00ad\b9\u00a4+\u00e8T}\u00bdR\u00ef\u0019\u00cb\u00af&M\u00e4&\u0080,\u008d\u0088WURe\u00c7d\u0000\u0000\u0000\u0002\u0003\u0001\u0000\u0000\nR\u0003\u0002\u0000\u0000\u001fA"
        }, 
        {
            "type": "HS_ADMIN", 
            "value": "\u000f\u00ff\u0000\u0000\u0000\t0.NA/0.NA\u0000\u0000\u0001,\u0000\u0000"
        }, 
        {
            "type": "HS_PUBKEY", 
            "value": "\u0000\u0000\u0000\u000bDSA_PUB_KEY\u0000\u0000\u0000\u0000\u0000\u0015\u0000\u0097`P\u008f\u0015#\u000b\u00cc\u00b2\u0092\u00b9\u0082\u00a2\u00eb\u0084\u000b\u00f0X\u001c\u00f5\u0000\u0000\u0000\u0081\u0000\u00fd\u007fS\u0081\u001du\u0012)R\u00dfJ\u009c.\u00ec\u00e4\u00e7\u00f6\u0011\u00b7R<\u00efD\u0000\u00c3\u001e?\u0080\u00b6Q&iE]@\"Q\u00fbY=\u008dX\u00fa\u00bf\u00c5\u00f5\u00ba0\u00f6\u00cb\u009bUl\u00d7\u0081;\u0080\u001d4o\u00f2f`\u00b7k\u0099P\u00a5\u00a4\u009f\u009f\u00e8\u0004{\u0010\"\u00c2O\u00bb\u00a9\u00d7\u00fe\u00b7\u00c6\u001b\u00f8;W\u00e7\u00c6\u00a8\u00a6\u0015\u000f\u0004\u00fb\u0083\u00f6\u00d3\u00c5\u001e\u00c3\u00025T\u0013Z\u0016\u00912\u00f6u\u00f3\u00ae+a\u00d7*\u00ef\u00f2\"\u0003\u0019\u009d\u00d1H\u0001\u00c7\u0000\u0000\u0000\u0081\u0000\u00f7\u00e1\u00a0\u0085\u00d6\u009b=\u00de\u00cb\u00bc\u00ab\\6\u00b8W\u00b9y\u0094\u00af\u00bb\u00fa:\u00ea\u0082\u00f9WL\u000b=\u0007\u0082gQYW\u008e\u00ba\u00d4YO\u00e6q\u0007\u0010\u0081\u0080\u00b4I\u0016q#\u00e8L(\u0016\u0013\u00b7\u00cf\t2\u008c\u00c8\u00a6\u00e1<\u0016z\u008bT|\u008d(\u00e0\u00a3\u00ae\u001e+\u00b3\u00a6u\u0091n\u00a3\u007f\u000b\u00fa!5b\u00f1\u00fbbz\u0001$;\u00cc\u00a4\u00f1\u00be\u00a8Q\u0090\u0089\u00a8\u0083\u00df\u00e1Z\u00e5\u009f\u0006\u0092\u008bf^\u0080{U%d\u0001L;\u00fe\u00cfI*\u0000\u0000\u0000\u0080o\u00fdA8\u00e5\u00c7b\u00af\u008da\u0080\u00da\t\u00802Ou&\u00a1K\u0014v\u00b1\u0092\u00f1\u00ba\u00cb\u00b7F\u00c6\u0019\u00d7\u00b1g(\u0095n\u00e1\u000b\u00e3\u000f\u00ecL\u00f6\u00c8\u00a8\u0093r{G\u00dc\u00f4\u00cd\u00bb'\u0014\u00a7\u00de\u0085\u00ee\u0087\u00b3\u0018~\u00e9\u0089\u0091G\n\u00af\u00ae\u009e-\u0005\tHf1\u00e6\r\u00fb{g\u009a\u00aa\u00efW\u00e1\u00d3\u00d7!\bc\u00a2\u00d2\u00f3U\u0087)\u00bf\u009f\u00a8(\u009a\u0096\u0010 Y\u0010\u00ee\n\u00a0d\u008b\u00ec\u00c91\u00dd\u000e\u00b00+e\u008e\u00d5m\u0091\u00f9"
        }, 
        {
            "type": "10320/sig.digest", 
            "value": "<digests hdl=\"0.NA/0.NA\">\n<val index=\"2\" md5=\"9B374AB1E8093C1FB2FD77A14A9F108C\" sha1=\"3E5CEB142ABBBEEA7FEF6102802E7BE701CE171C\"/>\n<val index=\"100\" md5=\"4C250C9C8256272A61A7BF7210DF04F6\" sha1=\"5E96B0569A7BDB9B8458CA6A0D5C3BAE7BDBE345\"/>\n<val index=\"300\" md5=\"57582B9CC8AAE3DB9386C61914802835\" sha1=\"72E9F139431A6AF01607931AD8374B7F65E537DD\"/>\n</digests>\n"
        }, 
        {
            "type": "10320/sig.sig", 
            "value": "<signature ofindex=\"400\" hdl=\"0.NA/0.NA\">\n<sig alg=\"SHA1withDSA\" signer=\"0.NA/0.NA\" signerIndex=\"300\">302C02146AD3CD6E09932432CC847DC90F2B91D1A51186F602144E4D7EF59946EC2D1780A119CA2E1B3F8B60154C</sig>\n</signature>\n"
        }
    ], 

...
    "0.NA/843": [
        {
            "type": "HS_SITE", 
            "value": "\u0000\u0001\u0002\u0005\u0000\u0001\u0080\u0002\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0001\u0000\u0000\u0000\u0004desc\u0000\u0000\u0000\"SURFsara handle & epic test server\u0000\u0000\u0000\u0001\u0000\u0000\u0000\u0001\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0091d\u001f\u009a\u0000\u0000\u0001\u00b8\u0000\u0000\u0000\u000bDSA_PUB_KEY\u0000\u0000\u0000\u0000\u0000\u0015\u0000\u0097`P\u008f\u0015#\u000b\u00cc\u00b2\u0092\u00b9\u0082\u00a2\u00eb\u0084\u000b\u00f0X\u001c\u00f5\u0000\u0000\u0000\u0081\u0000\u00fd\u007fS\u0081\u001du\u0012)R\u00dfJ\u009c.\u00ec\u00e4\u00e7\u00f6\u0011\u00b7R<\u00efD\u0000\u00c3\u001e?\u0080\u00b6Q&iE]@\"Q\u00fbY=\u008dX\u00fa\u00bf\u00c5\u00f5\u00ba0\u00f6\u00cb\u009bUl\u00d7\u0081;\u0080\u001d4o\u00f2f`\u00b7k\u0099P\u00a5\u00a4\u009f\u009f\u00e8\u0004{\u0010\"\u00c2O\u00bb\u00a9\u00d7\u00fe\u00b7\u00c6\u001b\u00f8;W\u00e7\u00c6\u00a8\u00a6\u0015\u000f\u0004\u00fb\u0083\u00f6\u00d3\u00c5\u001e\u00c3\u00025T\u0013Z\u0016\u00912\u00f6u\u00f3\u00ae+a\u00d7*\u00ef\u00f2\"\u0003\u0019\u009d\u00d1H\u0001\u00c7\u0000\u0000\u0000\u0081\u0000\u00f7\u00e1\u00a0\u0085\u00d6\u009b=\u00de\u00cb\u00bc\u00ab\\6\u00b8W\u00b9y\u0094\u00af\u00bb\u00fa:\u00ea\u0082\u00f9WL\u000b=\u0007\u0082gQYW\u008e\u00ba\u00d4YO\u00e6q\u0007\u0010\u0081\u0080\u00b4I\u0016q#\u00e8L(\u0016\u0013\u00b7\u00cf\t2\u008c\u00c8\u00a6\u00e1<\u0016z\u008bT|\u008d(\u00e0\u00a3\u00ae\u001e+\u00b3\u00a6u\u0091n\u00a3\u007f\u000b\u00fa!5b\u00f1\u00fbbz\u0001$;\u00cc\u00a4\u00f1\u00be\u00a8Q\u0090\u0089\u00a8\u0083\u00df\u00e1Z\u00e5\u009f\u0006\u0092\u008bf^\u0080{U%d\u0001L;\u00fe\u00cfI*\u0000\u0000\u0000\u0080=@;\u00c7\u00d9\u0005<\u00e1}\n\u00cb\u00d7\u0018\u00ccf\u00ed\u0018\u00cc!e\u00f5\u000f&\u00ee\u00f2\u001e\u008f\u0094&T\u00b7\u00b4~l\u009b,X\u00dcc\u00a0\u00af\u00dc\u00c8H\u0002\u007f\u0081e\u0013%X\u0094\u009a\u001bg\u0015m\u00c9K(\u00cd\u00df\u00ea\u0011\u00c28\u00a0?\u00f3\u00ac\u00e4\u00c4\u00a8\u00f6Yz\u00f1XeCXj\u00a2\u00fex]t>\u0010\u00b2\u001bh\\\u00d1j\u00ab9\u00c7H\u0095\u00ad\b9\u00a4+\u00e8T}\u00bdR\u00ef\u0019\u00cb\u00af&M\u00e4&\u0080,\u008d\u0088WURe\u00c7d\u0000\u0000\u0000\u0002\u0003\u0001\u0000\u0000\nR\u0003\u0002\u0000\u0000\u001fA"
        }, 
        {
            "type": "HS_ADMIN", 
            "value": "\fs\u0000\u0000\u0000\b0.NA/843\u0000\u0000\u0000\u00c8"
        }, 
        {
            "type": "HS_VLIST", 
            "value": "\u0000\u0000\u0000\u0004\u0000\u0000\u0000\t0.NA/0.NA\u0000\u0000\u0001,\u0000\u0000\u0000\b0.NA/843\u0000\u0000\u0001,\u0000\u0000\u0000\t843/ADMIN\u0000\u0000\u0000\u00c8\u0000\u0000\u0000\n843/ADMIN1\u0000\u0000\u0000\u00c8"
        }, 
        {
            "type": "HS_PUBKEY", 
            "value": "\u0000\u0000\u0000\u000bDSA_PUB_KEY\u0000\u0000\u0000\u0000\u0000\u0015\u0000\u0097`P\u008f\u0015#\u000b\u00cc\u00b2\u0092\u00b9\u0082\u00a2\u00eb\u0084\u000b\u00f0X\u001c\u00f5\u0000\u0000\u0000\u0081\u0000\u00fd\u007fS\u0081\u001du\u0012)R\u00dfJ\u009c.\u00ec\u00e4\u00e7\u00f6\u0011\u00b7R<\u00efD\u0000\u00c3\u001e?\u0080\u00b6Q&iE]@\"Q\u00fbY=\u008dX\u00fa\u00bf\u00c5\u00f5\u00ba0\u00f6\u00cb\u009bUl\u00d7\u0081;\u0080\u001d4o\u00f2f`\u00b7k\u0099P\u00a5\u00a4\u009f\u009f\u00e8\u0004{\u0010\"\u00c2O\u00bb\u00a9\u00d7\u00fe\u00b7\u00c6\u001b\u00f8;W\u00e7\u00c6\u00a8\u00a6\u0015\u000f\u0004\u00fb\u0083\u00f6\u00d3\u00c5\u001e\u00c3\u00025T\u0013Z\u0016\u00912\u00f6u\u00f3\u00ae+a\u00d7*\u00ef\u00f2\"\u0003\u0019\u009d\u00d1H\u0001\u00c7\u0000\u0000\u0000\u0081\u0000\u00f7\u00e1\u00a0\u0085\u00d6\u009b=\u00de\u00cb\u00bc\u00ab\\6\u00b8W\u00b9y\u0094\u00af\u00bb\u00fa:\u00ea\u0082\u00f9WL\u000b=\u0007\u0082gQYW\u008e\u00ba\u00d4YO\u00e6q\u0007\u0010\u0081\u0080\u00b4I\u0016q#\u00e8L(\u0016\u0013\u00b7\u00cf\t2\u008c\u00c8\u00a6\u00e1<\u0016z\u008bT|\u008d(\u00e0\u00a3\u00ae\u001e+\u00b3\u00a6u\u0091n\u00a3\u007f\u000b\u00fa!5b\u00f1\u00fbbz\u0001$;\u00cc\u00a4\u00f1\u00be\u00a8Q\u0090\u0089\u00a8\u0083\u00df\u00e1Z\u00e5\u009f\u0006\u0092\u008bf^\u0080{U%d\u0001L;\u00fe\u00cfI*\u0000\u0000\u0000\u0081\u0000\u00fc\u0010\u008c\u00ac\u00a9H\u00fc\u00fbj\u00e9\u00b0\u0093\u0083\u00b7\u00e1=P\u001e\"\u0088\u0089\u00b8ZC\u0086\n\u00c0\u00d6$\u0097\u00ec!\u00acY\u0015\u001f\u00de>:t]V\u00b1\u0080\u0015\f\u0010\u00b6<i\u00c4+\u00e0\u009d@\u00ad\u00ee,\u0080EV:\u009b\u00c3\u00b7\u00a5\u00da\u0019\u00ac\u00e0Dxx\u00fa\u009fH\"_\u0099;+J\u00ce\u0012>\u0016\b\u00e2\u00ad\u00a4uh+!hz\u008a\u00a9\u001a\u00cd;\u001c\u009f\u001c\u00af_\u00b0]\u009a_\u0006M\u00ce*\u001a!\u0001\u00aa(<\u00f0\u00aeJ\u0001\u00e6ro."
        }
    ], 

... 
}
```
We know that `HS_SECKEY` is present in `0.NA/843`. It is NOT shown.
